### PR TITLE
feat: iDontWant is sent only for gossipsub 1.2 or higher

### PR DESCRIPTION
**Pull request by diegomrsantos**
closes https://github.com/vacp2p/nim-libp2p/issues/1131

### CODE REVIEW

**Review by AlejandroCabeza — APPROVED** [review]
I think this test will fail when the GossipSubCodec is updated. It's fine to leave it like this and update it later, though.

**Comment by diegomrsantos** [comment]
> I think this test will fail when the GossipSubCodec is updated. It's fine to leave it like this and update it later, though.

My implementation is wrong, it should be sent for versions >= 1.2.

File: tests/pubsub/testgossipsub.nim [hunk 1]
```diff
@@ -871,6 +871,71 @@ suite "GossipSub":
...
+    await nodeB.switch.connect(
+      nodeC.switch.peerInfo.peerId, nodeC.switch.peerInfo.addrs
+    )
+
+    let bFinished = newFuture[void]()
+    proc handler(topic: string, data: seq[byte]) {.async.} =
+      discard
+
+    proc handlerB(topic: string, data: seq[byte]) {.async.} =
+      echo "handlerB ", topic, data.len
```
Comment (lchenut) [inline tests/pubsub/testgossipsub.nim:907]: I'm not a fan of leaving `echo` in tests. It might be interesting to replace it with a check, though.

Comment (diegomrsantos) [reply tests/pubsub/testgossipsub.nim:907]: oh thanks, I forgot it there.

**Review by lchenut — APPROVED** [review]

**Pull request merged**